cdap-data-fabric/src/test/java/co/cask/cdap/data2/metadata/dataset/MetadataDatasetTest.java

```diff
@@ -201,26 +201,38 @@ public void testSearchOnTags() throws Exception {
     Assert.assertEquals(0, dataset.getTags(stream1).size());
     dataset.addTags(app1, "tag1", "tag2", "tag3");
     dataset.addTags(flow1, "tag1");
-    dataset.addTags(dataset1, "tag3", "tag2", "tag12");
+    dataset.addTags(dataset1, "tag3", "tag2", "tag12-tag33");
     dataset.addTags(stream1, "tag2, tag4");
 
     // Try to search on all tags
     List<MetadataEntry> results =
-      dataset.searchByKeyValue("ns1", "tags:*", MetadataSearchTargetType.ALL);
+      dataset.search("ns1", "tags:*", MetadataSearchTargetType.ALL);
     Assert.assertEquals(4, results.size());
 
     // Try to search for tag1*
-    results = dataset.searchByKeyValue("ns1", "tags:tag1*", MetadataSearchTargetType.ALL);
+    results = dataset.search("ns1", "tags:tag1*", MetadataSearchTargetType.ALL);
     Assert.assertEquals(3, results.size());
 
     // Try to search for tag1
-    results = dataset.searchByKeyValue("ns1", "tags:tag1", MetadataSearchTargetType.ALL);
+    results = dataset.search("ns1", "tags:tag1", MetadataSearchTargetType.ALL);
     Assert.assertEquals(2, results.size());
 
     // Try to search for tag4
-    results = dataset.searchByKeyValue("ns1", "tags:tag4", MetadataSearchTargetType.ALL);
+    results = dataset.search("ns1", "tags:tag4", MetadataSearchTargetType.ALL);
     Assert.assertEquals(1, results.size());
 
+    // Try to search for tag33
+    results = dataset.search("ns1", "tags:tag33", MetadataSearchTargetType.ALL);
+    Assert.assertEquals(1, results.size());
+
+    // Try to search for tag33
+    results = dataset.search("ns1", "tag33", MetadataSearchTargetType.ALL);
+    Assert.assertEquals(1, results.size());
+
+    // Try to search for tag3
+    results = dataset.search("ns1", "tags:tag3*", MetadataSearchTargetType.ALL);
```

Comment:
perhaps add some verifications for a different namespace too, just for completeness